libgalois/include/katana/PropertyGraph.h

```diff
@@ -41,49 +56,53 @@ class KATANA_EXPORT GraphTopology {
 
   GraphTopology(
       LargeArray<Edge>&& adj_indices, LargeArray<Node>&& dests) noexcept
-      : adj_indices_(std::move(adj_indices)),
-        dests_(std::move(dests)),
-        out_indices(LargeToArrowArray(adj_indices_)),
-        out_dests(LargeToArrowArray(dests_)) {}
-
-  static std::unique_ptr<GraphTopology> Copy(
-      const GraphTopology& that) noexcept;
-
-  /*
-   * TODO(amber): consider re-enabling these in the near future
-  GraphTopology(GraphTopology&& that):
-    adj_indices_(std::move(that.adj_indices_)),
-    dests_(std::move(that.dests)),
-    out_indices(LargeToArrowArray(adj_indices_)),
-    out_dests(LargeToArrowArray(dests_))
-  {}
-
-  GraphTopology& operator = (GraphTopology&& that) {
+      : adj_indices_(std::move(adj_indices)), dests_(std::move(dests)) {}
+  // out_indices(LargeToArrowArray(adj_indices_)),
+  // out_dests(LargeToArrowArray(dests_)) {}
+
+  static GraphTopology Copy(const GraphTopology& that) noexcept;
+
+  GraphTopology(GraphTopology&& that)
+      : adj_indices_(std::move(that.adj_indices_)),
+        dests_(std::move(that.dests_)) {}
+
+  GraphTopology& operator=(GraphTopology&& that) {
     adj_indices_ = std::move(that.adj_indices_);
     dests_ = std::move(that.dests_);
-    out_indices = LargeToArrowArray(adj_indices_);
-    out_dests = LargeToArrowArray(dests_);
     return *this;
   }
-  */
 
   uint64_t num_nodes() const { return adj_indices_.size(); }
 
   uint64_t num_edges() const { return dests_.size(); }
 
-  bool Equals(const GraphTopology& other) const {
-    return out_indices->Equals(*other.out_indices) &&
-           out_dests->Equals(*other.out_dests);
+  const Edge* adj_data() const noexcept { return &adj_indices_[0]; }
+
+  const Node* dest_data() const noexcept { return &dests_[0]; }
+
+  bool Equals(const GraphTopology& that) const {
+    if (this == &that) {
+      return true;
+    }
+    if (this->num_nodes() != that.num_nodes()) {
```

Comment:
nit: some preference to omit `this` here as we use `num_nodes()`  (without this) everywhere else in this class

Reply:
Fixed: 6f4b8a0971e97a7fcad213f68d6e2d97359f777c